packages/protocol/lib/web3-utils.ts

```diff
@@ -237,113 +210,33 @@ export async function setInRegistryViaMultiSig(
-          name,
-          ...(await args(networkName))
-        )
-      })
-      .then(async () => {
-        if (then) {
-          return then(proxiedContract, web3, networkName)
-        }
-      })
-  }
-}
-
-/*
- * Used to deploy contracts whose Proxies have already been deployed.
- *
- * Arguments:
- * - artifacts: the Resolver object provided by Truffle
- * - name: name of the contract to deploy
- * - args: array of arguments to the contract's initializer
- * - then: a callback that can perform additional migration operations after deployment
- *
- * The callback will be called with the deployed proxied contract, a web3
- * instance (derived from the provider of the deployer given by Truffle), and the
- * name of the network (as given by Truffle).
- *
- * Returns:
- * A function with a signature as expected to be exported from a Truffle
- * migration script.
- */
-export function deployImplementationAndRepointProxy<
-  ContractInstance extends Truffle.ContractInstance
->(
+export function deployerForCoreContract<ContractInstance extends Truffle.ContractInstance>(
   web3: Web3,
   artifacts: any,
-  deployedProxyAddress: string,
-  name: string,
-  args: (networkName?: string) => Promise<any[]>,
+  name: CeloContract,
+  args: (networkName?: string) => Promise<any[]> = async () => [],
   then?: (contract: ContractInstance, web3: Web3, networkName: string) => void
 ) {
-  return async (deployer: any, networkName: string, _accounts: string[]) => {
+  const Contract = artifacts.require(name)
+  const ContractProxy = artifacts.require(name + 'Proxy')
+  return (deployer: any, networkName: string, _accounts: string[]) => {
     console.log('Deploying', name)
-    const Contract = artifacts.require(name)
-    const ContractProxy = artifacts.require(name + 'Proxy')
-    let proxiedContract: ContractInstance
-    deployer
-      // Hack to create build artifact.
-      .deploy(ContractProxy)
-      .then(async () => {
-        const artifact = ContractProxy._json
-        artifact.networks[await web3.eth.net.getId()] = {
-          address: deployedProxyAddress,
-          // @ts-ignore
-          transactionHash: '0x',
-        }
-        const contractsDir = build_directory + '/contracts'
-        const artifactor = new Artifactor(contractsDir)
-
-        await artifactor.save(artifact)
-      })
-      .then(() => {
-        return deployer.deploy(Contract)
-      })
-      .then(async () => {
-        proxiedContract = await setInitialProxyImplementation<ContractInstance>(
-          web3,
-          artifacts,
-          name,
-          ...(await args(networkName))
-        )
-      })
-      .then(async () => {
-        if (then) {
-          return then(proxiedContract, web3, networkName)
-        }
-      })
+    deployer.deploy(ContractProxy)
+    deployer.deploy(Contract)
+    deployer.then(async () => {
+      const proxy: ProxyInstance = await ContractProxy.deployed()
+      await proxy._transferOwnership(_accounts[0])
```

Comment:
Hmm, I can't remember why we did this. Shouldn't owner already be the first account?

Reply:
yeah, in fact many solidity contract transfer ownership on initialize, so this doesn't make a lot of sense.

but removing it sounded more controversial here.

Reply:
If it doesn't break migrations and you can verify that `_accounts[0]` is the same as the proxy's owner after migrating without this, I think it's fine to take out. With a clean up PR like this would be good to take out any confusing, unnecessary steps we've missed before like this.

Maybe reach out to @asaj as well, maybe he remembers if there was a reason for this?

Reply:
Created an issue for this
https://github.com/celo-org/celo-monorepo/issues/647